### PR TITLE
Remove Backbone dependancy

### PR DESCRIPTION
**Pull request by aron**
Now just using the Broadcast and Soak libraries for events and inheritance respectively. I think it would be best to bundle these two with the build as they are smaller libraries, leaving jQuery and underscore as external dependancies.

### CODE REVIEW

File: lib/m/module.js [hunk 1]
```diff
@@ -47,6 +61,10 @@
       return this;
     },
```
Comment (jeffchan) [inline lib/m/module.js:63]: Even though it's pretty self-explanatory, perhaps a simple comment to explain the `.$()` method? Same with `.initialize()`.

File: test/specs/module-spec.js [hunk 2]
```diff
@@ -797,9 +827,15 @@ describe('m.module()', function () {
       });
     });
 
+    describe('.initialize()', function () {
+      it('exists as a no-op', function () {
+        assert.isFunction(this.instance.teardown);
```
Comment (jeffchan) [inline test/specs/module-spec.js]: Should be `assert.isFunction(this.instance.initialize)` :)

Comment (aron) [inline test/specs/module-spec.js]: Good point :)

**Comment by aron** [comment]
Merged as of 0916f40

**Pull request closed without merging**